metabase/metabase
# Highlight selected events in the query builder

Epic: https://github.com/metabase/metabase/issues/20289

Changes:
- Clicking on the event icon in the chart highlights the event (group) and opens the sidebar
- Clicking again deselects the event (group)
- All timelines are available in the sidebar in the query builder
- Timelines within the question's collection are shown by default. As a result, timelines within `Our analytics` are shown for unsaved questions by default. We might want to fix this later.
- A timeline is expanded if it's the only available timeline or if there are highlighted events

How to test:
- Go to a question
- Click on the calendar icon in the bottom right corner and create events, if there are none
- Make sure they are displayed on the chart
- Click on the event and make sure it's highlighted and the sidebar is opened

<img width="1728" alt="Screenshot 2022-03-17 at 18 42 48" src="https://user-images.githubusercontent.com/8542534/158838025-37f8e0f5-9286-40bf-beb5-6a62692dff24.png">
